perf-tool/src/objectalloc.cpp

```diff
@@ -0,0 +1,150 @@
+                        JNIEnv* env,
+                        jthread thread,
+                        jobject object,
+                        jclass object_klass,
+                        jlong size) {
+    jvmtiError err;
+
+    json jObj;
+    char *classType;
+    auto start = steady_clock::now();
+
+
+    /*** get information about object ***/
+    err = jvmtiEnv->GetClassSignature(object_klass, &classType, NULL);
+    if (classType != NULL && err == JVMTI_ERROR_NONE) {
+        jObj["objType"] = classType;
+        jObj["size"] = (jint)size;
+    }
+
+
+    /*** get information about backtrace at object allocation sites if enabled***/
+    /*** retrieves method names and line numbers, and declaring class name and signature ***/
+    if (objAllocBackTraceEnabled) {
+        if (objAllocSampleCount % objAllocSampleRate == 0){
+            char *methodName;
+            char *methodSignature;
+            char *declaringClassName;
+            jclass declaring_class;
+            jint entry_count_ptr;
+            jvmtiLineNumberEntry* table_ptr;
+            jlocation start_loc_ptr;
+            jlocation end_loc_ptr;
+            int numFrames = 10;
+            jvmtiFrameInfo frames[numFrames];
+            jint count;
+            int i;
+            // int sze = 0;
+            auto jMethods = json::array();
+            err = jvmtiEnv->GetStackTrace(NULL, 0, numFrames, frames, &count);
+            if (err == JVMTI_ERROR_NONE && count >= 1) {
+                for (i = 0; i < count; i++) {
+                    json jMethod;
+                    err = jvmtiEnv->GetMethodName(frames[i].method, &methodName, &methodSignature, NULL);
+                    if (err == JVMTI_ERROR_NONE) {
+                        err = jvmtiEnv->GetMethodDeclaringClass(frames[i].method, &declaring_class);
+                        err = jvmtiEnv->GetMethodLocation(frames[i].method, &start_loc_ptr, &end_loc_ptr);
+                        err = jvmtiEnv->GetClassSignature(declaring_class, &declaringClassName, NULL);
+                        err = jvmtiEnv->GetLineNumberTable(frames[i].method, &entry_count_ptr, &table_ptr);
+                        if (err == JVMTI_ERROR_NONE) {
+                            jMethod["methodName"] = methodName;
+                            jMethod["methodSignature"] = methodSignature;
+                            jMethod["lineNum"] = table_ptr->line_number;
+                            jMethod["methodNum"] = i;
+                            jMethods.push_back(jMethod);
+                        }
+                    }
+                }
+            }
+
+            err = jvmtiEnv->Deallocate((unsigned char*)methodSignature);
+            err = jvmtiEnv->Deallocate((unsigned char*)methodName);
+            err = jvmtiEnv->Deallocate((unsigned char*)declaringClassName);
+
+            jObj["objBackTrace"] = jMethods;
+            jObj["objBackTraceSampleNum"] = objAllocSampleCount.load();
```

Comment:
This should receive the output of the `atomic_fetch_add` operation I mentioned above.
Also, I think this should be done irrespective of the stack tracing